simpeg/flow/richards/empirical.py

```diff
@@ -26,96 +25,35 @@ def _partition_args(mesh, Hcond, Theta, hcond_args, theta_args, **kwargs):
     if len(other_params) > 0:
         raise Exception("Unknown parameters: {}".format(other_params))
 
-    hcond = Hcond(mesh, **hcond_params)
-    theta = Theta(mesh, **theta_params)
+    hcond = Hcond(**hcond_params)
+    theta = Theta(**theta_params)
 
     return hcond, theta
 
 
-class NonLinearModel(props.HasModel):
-    """A non linear model that has dependence on the fields and a model"""
-
-    counter = None  #: A simpeg.utils.Counter object
-
-    def __init__(self, mesh, **kwargs):
-        self.mesh = mesh
-        super(NonLinearModel, self).__init__(**kwargs)
-
-    @property
-    def mesh(self):
-        return self._mesh
-
-    @mesh.setter
-    def mesh(self, value):
-        self._mesh = validate_type("mesh", value, discretize.base.BaseMesh, cast=False)
-
-    @property
-    def nP(self):
-        """Number of parameters in the model."""
-        return self.mesh.nC
-
-
-class BaseWaterRetention(NonLinearModel):
-    def plot(self, ax=None):
-        import matplotlib.pyplot as plt
-
-        if ax is None:
-            plt.figure()
-            ax = plt.subplot(111)
-
-        h = -np.logspace(-2, 3, 1000)
-        ax.semilogx(-h, self(h))
-        ax.set_title("Water retention curve")
-        ax.set_xlabel(r"Soil water potential, $-\psi$")
-        ax.set_ylabel("Water content, $\\theta$")
-
-
-class BaseHydraulicConductivity(NonLinearModel):
-    def plot(self, ax=None):
-        import matplotlib.pyplot as plt
-
-        if ax is None:
-            plt.figure()
-            ax = plt.subplot(111)
-
-        h = -np.logspace(-2, 3, 1000)
-        ax.loglog(-h, self(h))
-        ax.set_title("Hydraulic conductivity function")
-        ax.set_xlabel(r"Soil water potential, $-\psi$")
-        ax.set_ylabel("Hydraulic conductivity, $K$")
-
-
-class Haverkamp_theta(BaseWaterRetention):
-    theta_r, theta_rMap, theta_rDeriv = props.Invertible(
-        "residual water content [L3L-3]"
-    )
-
-    theta_s, theta_sMap, theta_sDeriv = props.Invertible(
-        "saturated water content [L3L-3]"
-    )
+class Haverkamp_theta(WaterRetention):
 
     alpha, alphaMap, alphaDeriv = props.Invertible("")
 
     beta, betaMap, betaDeriv = props.Invertible("")
 
     def __init__(
         self,
-        mesh,
+        *args,
```

Comment:
similarly here, could we add a comment to help developers who are learning by reading the code?